third_party/blink/renderer/platform/heap/member.h

```diff
@@ -91,6 +93,69 @@ struct MemberHash
   }
 };
 
+// Replay's hashing function with Member<>-wrapped objects, using the registered 
+// pointer id.  
+template <typename T>
+struct MemberHashRecordReplayRegisteredPointerId
+    : IntHash<cppgc::internal::MemberBase::RawStorage::IntegralType> {
+  using Base = IntHash<cppgc::internal::MemberBase::RawStorage::IntegralType>;
+  STATIC_ONLY(MemberHashRecordReplayRegisteredPointerId);
+  // Heap hash containers allow to operate with raw pointers, e.g.
+  //   HeapHashSet<Member<GCed>> set;
+  //   set.find(raw_ptr);
+  // Therefore, provide two hashing functions, one for raw pointers, another for
+  // Member. Prefer compressing raw pointers instead of decompressing Members,
+  // assuming the former is cheaper.
+  static unsigned GetHash(const T* key) {
+    int ptr = recordreplay::PointerId(key);
+    CHECK(ptr != 0);
```

Comment:
I think this check will fail when we're not recording or replaying.  Can we test for `IsRecordingOrReplaying("pointer-ids")` like we do in the removed CompareMemberByRecordReplayId, and use the a regular pointer hash if that's not the case?  This both covers the non-recording/replaying case, and also allows us to switch back to the regular hashing strategy by selectively disabling the "pointer-ids" feature (this can be done via the environment) so we can more easily see how the changes we've made affect the browser's performance / correctness.